src/controllers/audit-controller.ts

```diff
@@ -0,0 +1,54 @@
+import { getRepository, Repository } from 'typeorm';
+import { Audit } from '../entities/audit-entity';
+import { User } from '../entities/user-entity';
+import { AuditEventsAction } from '../entities/audit-action-events-entity';
+import { RefreshToken } from '../entities/auth-entity';
+import { logger } from '../configs/logger';
+
+class AuditController {
+  private audits: Repository<Audit>;
+
+  private auditEvents: Repository<AuditEventsAction>;
+
+  private users: Repository<User>;
+
+  private tokens: Repository<RefreshToken>;
+
+  public constructor() {
+    this.audits = getRepository(Audit);
+    this.auditEvents = getRepository(AuditEventsAction);
+    this.users = getRepository(User);
+    this.tokens = getRepository(RefreshToken);
+  }
+
+  private getUser = async (token: string, id: string): Promise<User> => {
+    let userId = id;
+
+    if (!userId) {
+      const tokenRecord = await this.tokens.findOne({ token });
+
+      if (tokenRecord) {
+        userId = tokenRecord.userId;
+      }
+    }
+
+    return (await this.users.findOne({ id: userId })) as User;
```

Comment:
Hi, Dima! Please get rid of such contructions, because it twice process of awaiting.
It's only one exclusion in this but you have to see difference:
```js
return (await this.users.findOne({id})).getFullName();
```